org.jacoco.core.test/src/org/jacoco/core/internal/flow/InstructionTest.java

```diff
@@ -58,6 +59,13 @@ public void testSetPredecessor() {
 				new InsnNode(Opcodes.NOP), 122);
 		instruction.setPredecessor(predecessor, 0);
 		assertEquals(1, predecessor.getBranches());
+
+		try {
+			instruction.setPredecessor(predecessor, 0);
+			fail("exception expected");
+		} catch (IllegalStateException e) {
```

Comment:
@Godin Ah I see. But in this case I would prefer to have a separate test case specifically for this scenario.